qiskit/transpiler/passes/ms_basis_decomposer.py

```diff
@@ -0,0 +1,102 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Pass for converting a circuit targeting U3,CX basis to Rx,Ry,Rxx."""
+
+from qiskit.transpiler.basepasses import TransformationPass
+from qiskit.exceptions import QiskitError
+
+from qiskit.converters import circuit_to_dag
+from qiskit.extensions.standard import U3Gate, CnotGate
+
+from qiskit.transpiler.passes import Unroller
+from qiskit.quantum_info.synthesis.one_qubit_decompose import OneQubitEulerDecomposer
+from qiskit.quantum_info.synthesis.ion_decompose import cnot_rxx_decompose
+
+
+class MSBasisDecomposer(TransformationPass):
+    """
+    Convert a circuit in U3,CX to Rx,Ry,Rxx without unrolling or simplification.
+    """
+
+    supported_input_gates = (U3Gate, CnotGate)
+    supported_basis_names = ('rx', 'ry', 'rxx', 'ms')
+
+    def __init__(self, basis):
```

Comment:
In many part of terra, we refer to the basis set as `basis_gates`. Could be this renamed in that way, for consistency?